platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/ReconnectNodeRemover.java

```diff
@@ -115,21 +114,14 @@ public ReconnectNodeRemover(
     public void setPathInformation(final long newFirstLeafPath, final long newLastLeafPath) {
```

Comment:
Good point. Method name still corresponds to what the method does, it updates this node remover's leaf path range. The added code is a minor optimization: rather than remove all old leaves lesser than the new first leaf path one by one, all of them are removed here. I added a paragraph about it to the JavaDoc.

Reply:
This method is merged with `newInternalNode` which was previously `syncrhonized`. Now this method should be too.